### DISCUSSION TITLE
Steering Council meeting minutes

### DISCUSSION THREAD

**Discussion by jGaboardi** [pysal/pysal#1300] [Announcements]
This thread will serve as the running documentation for the Steering Council meeting minutes, like the one for the [monthly developer meetings](https://github.com/orgs/pysal/discussions/1278). The previous minutes archive are [here](https://hackmd.io/YCjAJDgOTlKMpcN4H9P2UA).

**Comment by jGaboardi** [comment]
# 2024-02-07

## Attendance
* @knaaptime 
* @jGaboardi 
* @martinfleis 

## Agenda

*  Future of the [Code of Conduct Committee](https://github.com/pysal/governance/blob/main/conduct/code_of_conduct.rst#reporting-guidelines)
   * see also [`pysal/code_of_conduct`](https://github.com/pysal/code_of_conduct)
   * @martinfleis mentioned NumFocus during the [2024-01 meeting](https://github.com/orgs/pysal/discussions/1278#discussioncomment-8005392)
* Documentation (see  https://github.com/pysal/pysal/issues/1282)
* etc...

**Reply by knaaptime** [reply]
- versioning on docs
- standardization of user guide/notebooks project
- decide on common (modern) theme

**Reply by knaaptime** [reply]
- reconsider doc publishing recipe to point to last release rather than current `main`

**Reply by jGaboardi** [reply]
Code of Conduct notes:
* NumFocus summit in Amsterdam where potential for NF hosted CoC was discussed
* CoC infrastructure survey went out in 2023-12
* CoC planning team working on survey report in 2024-01
* Will this be available for *affiliated* projects or only *sponsored* projects?
* Martin will keep a look out.

**Comment by jGaboardi** [comment]
# 2025-02-06

## Attendance

* @jGaboardi 
* @sjsrey 
* @knaaptime 
* @gegen07 


## Agenda

* gsoc 
   * ideas
      * `momepy` – street-based tessellation
         * @gegen07 has some code on that
      * `spopt` – TSP
         * See https://github.com/pysal/spopt/issues/464 & https://github.com/pysal/spopt/pull/465
      * Multiple Dispatch
         * `pointpats`, `spreg`, etc... 
   * @sjsrey will crunch over the weekend and ping on Sunday  
* esda
   * do a last-minute release?
   * decision -- *yes* 
* meta
   * changelog automation issues 
   * potentially a composite workflow built of micro-actions
   * "PR"  (public relations) highlights for individual contributions
      * create issues for each package in PySAL
      * tag package maintainer for highlights
      * potentially automate this 
* spaghetti
   * [stagnant since 2018](https://github.com/pysal/spaghetti/discussions/195)
   * will break once geometries from `libpysal.cg` are removed
      * https://github.com/pysal/spaghetti/issues/646 
   * needs love or decision to archive
   * Decision
      * move core functionality into `lib`
      * can be directly consumed by `segregation`, `momepy`, etc.
      * sunset `spaghetti` like what we did with `splot` & `spvcm`

**Comment by jGaboardi** [comment]
# 2025-05-01

## Attendance

* @gegen07 
* @knaaptime 
* @jGaboardi 


## Agenda

* Proposed CoC by NumFocus
   * https://numfocus.org/code-of-conduct
   * Rec. passed unanimously by Steering Council
* GSoC update (@germano)
  * 2 slots
  * `spopt`
     * routing
     * flow-refueling model

**Comment by jGaboardi** [comment]
# 2025-08-21

## Attendance

* @gegen07 
* @knaaptime 
* @jGaboardi 

## Agenda

* GSoC update - Germano
  * both projects successful 
*  NSF grant update (Eli)
  * NARSC session
  * AAG co-organized session?

**Comment by jGaboardi** [comment]
# 2026-02-19

## Attendance

* James
* Germano
* Eli
* Martin

## Agenda

* Upcoming BIDS summit and AAG 2026
* PySAL policy on contributions by & interactions with LLMs
   * https://github.com/mhoye/maintenance-terms
   * https://github.com/melissawm/open-source-ai-contribution-policies
   * Discussion points
     * Create / add to Contribution Guidelines...
        * "humans + communications w/ other humans is our #&#8203;1 resource"
        * "massive drain on maintainer time & energy to interact with bots"
        * "we reserve the right to ignore these interactions and close opened issues & PRs"
        * "the steering council is united in this"
* Future of GSoC wrt current situation with LLMs?

**Comment by jGaboardi** [comment]
# 2026-05-21

## Attendance

* James
* Germano
* Eli

## Agenda

* Process to nominate a new pysal dev
  * current dev must nominate 
  * nominee must have a positive track record of engagement with the ecosystem and participation with community. This can include:
     * code/issues
     * meetings
     * federated package contribution
* Process to add new packages to the ecosystem
  * @knaaptime's idea (see below)
* More discussion on AI contribution policy 
  * Values (?)
     * transparent
     * disclosure
     * accountability 
  * Contribution Guidelines (not even AI specific)
     * If the contributor is not on the core term and/or more than XXX files or YYY lines are touched, an issues must be open to discuss before the PR 
     * Basically --> good PR hygiene

### Eli's idea

#### Criteria for new pysal package:

##### Round 1

1. is it germane to  pysal?
2. spatial analysis?
3. scientific contribution?
4. is it distinct?
    * warrant a new package or should proposed functionality be absorbed into existing packages
5. is the dependency surface acceptable?
6. is the license compatible
7. is there capacity? (conditional)
   * 2 devs, according to policy

##### Round 2

1. if the proposed package meets first two, it passes steering committee and goes to floor at discussion; requires at least two volunteers to maintain, then goes to floor to vote (core maintainers have voting privilege)

**Comment by sjsrey** [comment]
# 2026-07-15

## Attendance

## Agenda

- pysal LLM Policy
  - [networkx policy](https://networkx.org/documentation/latest/developer/contribute.html)
  - [oxide](https://discord.com/channels/1192517762103398401/1202659841072431255/1525572899644117102)
- ecosystem documentation standardization project
   - [quantecon](https://quantecon.org/lectures/)
   - [diataxis](https://diataxis.fr/) 
- pysal leadership transition plan

**Reply by knaaptime** [reply]
- we like the idea of adopting NX's policy, giving credit to their org for drafting the original
	- will draft, pre-approve in the SC, solicit input from the dev team, then take a vote
- we will standardize structure for subapackage docs that conforms to the diataxis plan
- transition plan will be roughly parlaimentary; we need a rotating prime minister who will also be in charge of the meta release
